### PR TITLE
Fixes nuget.exe update issue with not deleting old content files

### PR DESCRIPTION
**Pull request by deepakaravindr**
Fixes NuGet/Home#1498

On remove file, we are passing the path relative to the project directory
to DeleteFile method. It should be the full path.

Added a test to determine that the files are deleted as needed

@yishaigalatzer @danliu @MeniZalzman @feiling

### CODE REVIEW

**Comment by danliu** [comment]
:shipit:

**Comment by yishaigalatzer** [comment]
@deepakaravindr Looks like we only have one pretty complicated test for the scenario? Isn't it possible to write unit tests for `MSBuildProjectSystem.cs` ?

**Comment by yishaigalatzer** [comment]
:watch:

**Comment by MeniZalzman** [comment]
:watch:

**Comment by feiling** [comment]
:shipit:

**Comment by deepakaravindr** [comment]
@johnataylor

File: test/NuGet.CommandLine.Test/Util.cs [hunk 1]
```diff
@@ -486,5 +504,71 @@ public static JObject CreateSinglePackageRegistrationBlob(MockServer server, str
...
+            string projectName = "proj1",
+            string targetFrameworkVersion = "v4.5",
+            string[] references = null,
+            string[] contentFiles = null)
+        {
+            var referencesFormat = "<Reference Include='{0}' />";
+
+            var contentFileFormat = @"<Content Include = '{0}' />";
+
+            var projFileFormat = @"<Project ToolsVersion='4.0' DefaultTargets='Build'
```
Comment (johnataylor) [inline test/NuGet.CommandLine.Test/Util.cs]: don't use raw string concat for building XML - even in test code.
XML object models are easier to write, read and update (and less fragile)
(make sure to use System.Xml.Linq)

File: test/NuGet.CommandLine.Test/NuGetUpdateCommandTests.cs [hunk 2]
```diff
@@ -506,5 +437,123 @@ public class NuGetUpdateCommandTests
...
+                {
+                    var packagesConfigReader = new PackagesConfigReader(packagesConfigStream);
+                    var packages = packagesConfigReader.GetPackages().ToList();
+                    Assert.Equal(1, packages.Count);
+                    Assert.Equal(a1, packages[0].PackageIdentity);
+                }
+
+                // Act
+                var r = CommandRunner.Run(
+                    Util.GetNuGetExePath(),
```
Comment (johnataylor) [inline test/NuGet.CommandLine.Test/NuGetUpdateCommandTests.cs:535]: Its not clear sometimes what is test code and what isn't. In particular the name Util is deceptive.

**Comment by johnataylor** [comment]
:shipit:

**Pull request merged**